### PR TITLE
Sample gen for MPG

### PR DESCRIPTION
**Pull request by MaryGao**

### CODE REVIEW

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/operationsListSample.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+async function operationsListMaximumSetGen(): void {
+  const credential = new DefaultAzureCredential();
+  const client = new NetworkAnalyticsClient(credential);
+  const result = await client.operations.list();
+  console.log(result);
+}
+
+async function operationsListMaximumSetGenGeneratedByMinimumSetRuleMinimumSetGen(): void {
+  const credential = new DefaultAzureCredential();
+  const client = new NetworkAnalyticsClient(credential);
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/operationsListSample.ts:15]: FIXME

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/operationsListSample.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { NetworkAnalyticsClient } from "../src/networkAnalyticsClient.js";
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/operationsListSample.ts:4]: FIXME

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { NetworkAnalyticsClient } from "../src/networkAnalyticsClient.js";
+
+async function dataProductsAddUserRoleMaximumSetGen(): void {
+  const credential = new DefaultAzureCredential();
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts:7]: FIXEME for import

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  const subscriptionId = "00000000-0000-0000-0000-00000000000";
+  const client = new NetworkAnalyticsClient(credential, subscriptionId);
+  const result = await client.dataproducts.addUserRole(
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts:10]: FIXME for LRO

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsListByResourceGroupSample.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  );
+  console.log(result);
+}
+
+async function dataProductsListByResourceGroupMaximumSetGenGeneratedByMinimumSetRuleMinimumSetGen(): void {
+  const credential = new DefaultAzureCredential();
+  const subscriptionId = "00000000-0000-0000-0000-00000000000";
+  const client = new NetworkAnalyticsClient(credential, subscriptionId);
+  const result = await client.dataproducts.listByResourceGroup(
+    "aoiresourceGroupName",
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsListByResourceGroupSample.ts:21]: FIMX for parameter ordering

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      userName: "UserName",
+      dataTypeScope: ["scope"],
+      principalType: "User",
+      role: "Reader",
+    },
+  );
+  console.log(result);
+}
+
+async function dataProductsAddUserRoleMaximumSetGenGeneratedByMinimumSetRuleMinimumSetGen(): void {
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts:25]: FIXME add documents

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { NetworkAnalyticsClient } from "../src/networkAnalyticsClient.js";
+
+/**
+ * This sample demonstrates how to assign role to the data product.
+ *
+ * @summary assign role to the data product.
+ * x-ms-original-file: 2023-11-15/DataProducts_AddUserRole_MaximumSet_Gen.json
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts:10]: FIXME confirm with wd for the file issue

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsCreateSample.ts [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+ * This sample demonstrates how to create data product resource.
+ *
+ * @summary create data product resource.
+ * x-ms-original-file: 2023-11-15/DataProducts_Create_MaximumSet_Gen.json
+ */
+async function dataProductsCreateMaximumSetGen(): void {
+  const credential = new DefaultAzureCredential();
+  const subscriptionId = "00000000-0000-0000-0000-00000000000";
+  const client = new NetworkAnalyticsClient(credential, subscriptionId);
+  const result = await client.dataproducts.create(
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsCreateSample.ts:16]: FIXME fix client-level method

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsCreateSample.ts [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+    "aoiresourceGroupName",
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsCreateSample.ts:17]: FIXME fix nested operation group

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts [hunk 10]
```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { NetworkAnalyticsClient } from "@azure/arm-networkanalytics";
+import { DefaultAzureCredential } from "@azure/identity";
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts:5]: update net enc config https://github.com/Azure/azure-sdk-for-js/issues/30263

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts [hunk 11]
```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { NetworkAnalyticsClient } from "../src/networkAnalyticsClient.js";
```
[the remark this answers is not in the document]
Comment (MaryGao) [reply packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts]: create issues for sample build step

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/package.json [hunk 12]
```diff
@@ -93,136 +74,5 @@
...
-      },
-      "require": {
-        "types": "./dist/commonjs/api/dataProducts/index.d.ts",
-        "default": "./dist/commonjs/api/dataProducts/index.js"
-      }
-    }
-  },
-  "main": "./dist/commonjs/index.js",
-  "types": "./dist/commonjs/index.d.ts"
+  }
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/package.json:77]: update sample metadata

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (MaryGao) [reply packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts]: 1. remove useless env considering we don't need to run sample really
1. update package.json
`"build:samples": "dev-tool samples publish --force",` && `"execute:samples": "dev-tool samples run samples-dev",`

[the remark this answers is not in the document]
Comment (kazrael2119) [reply packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts]: before run ```rushx build samples```, need do some extra things:
1. add 
```
"//sampleConfiguration": {
    "productName": "",
    "productSlugs": [
      "azure"
    ],
    "disableDocsMs": true,
    "apiRefLink": "https://docs.microsoft.com/javascript/api/@azure/arm-networkanalytics?view=azure-node-preview"
  },
```
in package.json

2. create a ```sample.env``` file with
```
# App registration secret for AAD authentication
AZURE_CLIENT_SECRET=
AZURE_CLIENT_ID=
AZURE_TENANT_ID= 
```

then we can publish samples as well

[the remark this answers is not in the document]
Comment (kazrael2119) [reply packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataproductsAddUserRoleSample.ts]: error in excute:samples:
image

File: packages/typespec-ts/src/index.ts [hunk 13]
```diff
@@ -60,6 +60,7 @@ import { GenerationDirDetail, SdkContext } from "./utils/interfaces.js";
 import { provideContext, useContext } from "./contextManager.js";
 import { emitSerializerHelpersFile } from "./modular/buildHelperSerializers.js";
 import { provideSdkTypes } from "./framework/hooks/sdkTypes.js";
+import { buildSamples as buildModularSamples } from "./next/buildSamples.js";
```
Comment (MaryGao) [inline packages/typespec-ts/src/index.ts:63]: Enable UT

File: packages/typespec-ts/test/integration/azureExampleClient.spec.ts [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+} from "./generated/azure/example/basic/src/index.js";
+import { assert } from "chai";
+describe("Basic Rest Client", () => {
+  let client: BasicClient;
+
+  beforeEach(() => {
+    client = BasicClientFactory({ allowInsecureConnection: true });
+  });
+
+  it("basic example client action", async () => {
```
Comment (MaryGao) [inline packages/typespec-ts/test/integration/azureExampleClient.spec.ts:12]: pls remove rlc sample gen for now.

**Pull request closed without merging**